### PR TITLE
Expose eager c_api_experimental to other language binding.

### PR DESCRIPTION
**Pull request by Oceania2018**

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
I thought the experimental C API is private only, shouldn't be dependent on.

Adding @gunan to confirm

**Comment by Oceania2018** [comment]
It's exposed for non-eager mode. https://github.com/tensorflow/tensorflow/blob/874adafd91b0719b89ed1b77cfc472251a82d36b/tensorflow/BUILD#L701

The point is the [tf.net](https://github.com/SciSharp/TensorFlow.NET) need some of them.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by Oceania2018** [comment]
https://github.com/tensorflow/tensorflow/pull/37803/files

File: tensorflow/c/eager/c_api_experimental.cc [hunk 1]
```diff
@@ -31,6 +31,7 @@ using tensorflow::string;
 void TFE_OpReset(TFE_Op* op_to_reset, const char* op_or_function_name,
                  const char* raw_device_name, TF_Status* status) {
   if (op_to_reset) {
+    op_to_reset->operation->Clear();
```
Comment (mihaimaruseac) [inline tensorflow/c/eager/c_api_experimental.cc]: This is unrelated and should be in a separate PR

**Review by mihaimaruseac — APPROVED** [review]

**Comment by Oceania2018** [comment]
You guys are efficient. @mihaimaruseac

**Review by jaingaurav — APPROVED** [review]

**Comment by jaingaurav** [comment]
> I thought the experimental C API is private only, shouldn't be dependent on.
> 
> Adding @gunan to confirm

@mihaimaruseac: Experimental APIs are fine to call with the caveat that they have no stability guarantees.

**Review by alextp — APPROVED** [review]
